src/geo_key_directory.rs

```diff
@@ -0,0 +1,658 @@
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjScaleAtNatOrigin => {
+                    directory.proj_scale_at_nat_origin = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjScaleAtCenter => {
+                    directory.proj_scale_at_center = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjAzimuthAngle => {
+                    directory.proj_azimuth_angle = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjStraightVertPoleLong => {
+                    directory.proj_straight_vert_pole_long = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::Vertical => {
+                    directory.vertical =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+                GeoKeyDirectoryTag::VerticalCitation => {
+                    directory.vertical_citation = Self::get_string(
+                        &ascii_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::VerticalDatum => {
+                    directory.vertical_datum =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+                GeoKeyDirectoryTag::VerticalUnits => {
+                    directory.vertical_units =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+            }
+        }
+
+        Ok(directory)
+    }
+
+    fn get_short(
+        key_tag: GeoKeyDirectoryTag,
+        location_tag: Option<Tag>,
+        count: u16,
+        offset: u16,
+    ) -> TiffResult<u16> {
+        if location_tag.is_some() {
```

Comment:
Suggestion based on https://github.com/georust/geotiff/pull/19#discussion_r1769714908

```suggestion
        // Check that TIFFTagLocation == 0 so value is of SHORT type
        if location_tag.is_some() {
```